llama-index-integrations/readers/llama-index-readers-google/llama_index/readers/google/docs/base.py

```diff
@@ -148,9 +156,76 @@ def _read_structural_elements(self, elements: List[Any]) -> Any:
                 text += self._read_structural_elements(toc.get("content"))
         return text
 
+    def _determine_heading_level(self, element):
+        """Extracts the heading level, label, and ID from a document element.
+
+        Args:
+            element: a Structural Element.
+        """
+        level = None
+        heading_label = None
+        heading_id = None
+        if self.split_on_heading_level and "paragraph" in element:
+            style = element.get("paragraph").get("paragraphStyle")
+            style_type = style.get("namedStyleType", "")
+            heading_id = style.get("headingId", None)
+            if style_type == "TITLE":
+                level = 0
+                heading_label = "title"
+            elif style_type.startswith("HEADING_"):
+                level = int(style_type.split("_")[1])
+                if level > self.split_on_heading_level:
+                    return None, None, None
+
+                heading_label = f"heading_{level}"
+
+        return level, heading_label, heading_id
+
+    def _structural_elements_to_docs(
+        self, elements: List[Any], doc_metadata: dict
+    ) -> Any:
+        """Recurse through a list of Structural Elements.
+
+        Split documents on heading if split_on_heading_level is set.
+
+        Args:
+            elements: a list of Structural Elements.
+        """
+        docs = []
+
+        current_heading_level = self.split_on_heading_level
+
+        metadata = doc_metadata.copy()
+        text = ""
+        for value in elements:
+            element_text = self._read_structural_elements([value])
+
+            level, heading_label, heading_id = self._determine_heading_level(value)
+
+            if level is not None:
+                if level == self.split_on_heading_level:
+                    if text.strip():
+                        docs.append(Document(text=text, metadata=metadata.copy()))
+                        text = ""
+                    if heading_id:
+                        metadata["heading_id"] = heading_id
+                elif level < current_heading_level:
+                    metadata = doc_metadata.copy()
+
+                metadata[heading_label] = element_text
+                current_heading_level = level
+            else:
+                text += element_text
+
+        if text:
+            docs.append(Document(text=text, metadata=metadata))
```

Comment:
can we make sure to set `id_` like before?

Also, can  we update the readme to document these new options? (also a good time to update the readme to reflect v0.10.x)

Reply:
I added support for setting the `id_` to the document_id like before. If headers are supported the header_id or a random id is appended to the document_id. 

About the readme it seems to be empty. As I haven't used the other readers in the google module I could either add documentation about only the GoogleDocsReader or break it out to its own module. What do you think?